### PR TITLE
refactor: updating the project 

### PR DESCRIPTION
**Pull request by Dudleyneedham**
# Refactor the transaction

Updating the current project to split the different functions and reduce the noise in the `transaction.ts` file.

Moving things to the utils that could be used again.

If the types will be exported, move it to its file in the `utils` dir.

Renaming components to align with React convention.

### CODE REVIEW

**Comment by Dudleyneedham** [comment]
I will continue to work on the app refactor. Additionally, I will update the eslint and styles to match the current apps implementation instead of changing it.

**Comment by Dudleyneedham** [comment]
I have updated the eslint and added prettier as I noticed some places had trailing semi-colons, and others did not. Additionally, I wanted to make it easier for others who use the project without relying on their settings in the editor and mixing and matching with the project owners. I tried my best to match Alister's current settings. 

I sorted more utility functions into separate files and merged some similar files into a more uniform file, e.g. formatting, and will move the `utils/balances` into a singular file.

**Review by alistair-singh — CHANGES_REQUESTED** [review]
Looks good. Minor comments.

**Comment by alistair-singh** [comment]
Preview builds failed. Seems related to the case of files. Need to commit and push the rename explicitly with git.
```
[22:19:46.572]   ▲ Next.js 14.2.3
[22:19:46.572] 
[22:19:46.650]    Creating an optimized production build ...
[22:20:16.568] Failed to compile.
[22:20:16.569] 
[22:20:16.569] ./app/history/page.tsx
[22:20:16.569] Module not found: Can't resolve '@/components/ErrorDialog'
[22:20:16.570] 
[22:20:16.570] https://nextjs.org/docs/messages/module-not-found
[22:20:16.570] 
[22:20:16.570] ./app/status/page.tsx
[22:20:16.570] Module not found: Can't resolve '@/components/ErrorDialog'
[22:20:16.570] 
[22:20:16.570] https://nextjs.org/docs/messages/module-not-found
[22:20:16.570] 
[22:20:16.570] ./app/layout.tsx
[22:20:16.570] Module not found: Can't resolve '@/components/BridgeStatus'
[22:20:16.570] 
[22:20:16.570] https://nextjs.org/docs/messages/module-not-found
[22:20:16.570] 
[22:20:16.570] ./app/layout.tsx
[22:20:16.570] Module not found: Can't resolve '@/components/Footer'
[22:20:16.570] 
[22:20:16.570] https://nextjs.org/docs/messages/module-not-found
[22:20:16.570] 
[22:20:16.570] ./app/layout.tsx
[22:20:16.570] Module not found: Can't resolve '@/components/Menu'
[22:20:16.570] 
[22:20:16.570] https://nextjs.org/docs/messages/module-not-found
[22:20:16.570] 
[22:20:16.579] 
[22:20:16.579] > Build failed because of webpack errors
[22:20:16.588]  ELIFECYCLE  Command failed with exit code 1.
[22:20:16.610] Error: Command "pnpm run build" exited with 1
[22:20:17.042] 
```

File: components/ui/accordion.tsx [hunk 1]
```diff
@@ -4,7 +4,7 @@ import * as React from "react";
 import * as AccordionPrimitive from "@radix-ui/react-accordion";
 import { ChevronDown } from "lucide-react";
 
-import { cn } from "@/lib/utils";
+import { cn } from "@/utils/cn";
```
[the remark this answers is not in the document]
Comment (Dudleyneedham) [reply components/ui/accordion.tsx:7]: Ok, I haven't used shadon before, but I will move it back. I thought someone had made it.

File: components/ui/avatar.tsx [hunk 2]
```diff
@@ -1,9 +1,9 @@
-"use client"
+"use client";
```
[the remark this answers is not in the document]
Comment (Dudleyneedham) [reply components/ui/avatar.tsx:1]: I will revert the changes. I was wondering why the formatting was different only in the UI components.

File: eslint.config.js [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+const compat = new FlatCompat({
+  baseDirectory: __dirname,
+  recommendedConfig: js.configs.recommended,
+  allConfig: js.configs.all,
+});
+
+export default [
+  ...compat.extends("next/core-web-vitals", "prettier"),
+  {
+    ignores: [".next/**/*"],
```
[the remark this answers is not in the document]
Comment (Dudleyneedham) [reply eslint.config.js]: I added it in both prettier and eslint

**Review by alistair-singh — APPROVED** [review]

**Pull request merged**